### PR TITLE
Add GitHub build action

### PR DESCRIPTION
**Pull request by olemke**

### CODE REVIEW

File: .github/workflows/pythonapp.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.7
```
Comment (lkluft) [inline .github/workflows/pythonapp.yml]: Is it possible to check several Python versions?

File: .github/workflows/pythonapp.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        pip install Cython
+        pip install GDAL==$(gdal-config --version) --global-option=build_ext --global-option="-I/usr/include/gdal -L/usr/local/lib -lgdal"
+        pip install .[tests]
+    - name: Lint with flake8
+      run: |
+        pip install flake8
+        # stop the build if there are Python syntax errors or undefined names
+        # flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+        flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
```
Comment (lkluft) [inline .github/workflows/pythonapp.yml]: Does the build fail, when `flake8` returns with a non-zero exit code? If yes, is there an "allow failure" option?

File: .github/workflows/pythonapp.yml (hunk 1 above)
Comment (olemke) [reply .github/workflows/pythonapp.yml]: Yep, it's now testing on 3.6-3.8.

File: .github/workflows/pythonapp.yml (hunk 2 above)
Comment (olemke) [reply .github/workflows/pythonapp.yml]: It does. Unfortunately, the github action does not give a visual indication when continue-on-error is enabled and the test failed. It shows the same green checkmark as for a successful test.

Comment (olemke) [reply .github/workflows/pythonapp.yml]: Oh, but it does show an annotation indicating the failure. So that's good.

**Review by lkluft — APPROVED** [review]

File: .github/workflows/build-test-windows.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      with:
+        fetch-depth: 1
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install .
```
Comment (lkluft) [inline .github/workflows/build-test-windows.yml:26]:
```suggestion
        pip install .[tests]
```

Comment (olemke) [reply .github/workflows/build-test-windows.yml:26]: No, that doesn't work in Windows. No way to install gdal.

**Review by lkluft — APPROVED** [review]

**Pull request merged**